### PR TITLE
Add recovery password in the classic signIn

### PR DESCRIPTION
**Pull request by brunojdo**
#### What is the purpose of this pull request?

Add recovery password in the classic signIn

**Secondary** Two first commits are minor issues. 

#### Screenshots or example usage
https://login--storecomponents.myvtexdev.com/

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/EmailAndPassword.js [hunk 1]
```diff
@@ -154,12 +156,12 @@ class EmailAndPassword extends Component {
                 onClick={e => this.handleOnSubmit(e)}
                 isLoading={isLoading}
               >
-                <span className="f7">{translate(send, intl)}</span>
+                <span className="f7">{translate('login.signIn', intl)}</span>
```
Comment (salesfelipe) [inline react/components/EmailAndPassword.js]: Why don´t you use the <FormattedMessage id="..." /> instead of the translate?

File: react/Login.js [hunk 2]
```diff
@@ -100,21 +107,18 @@ class Login extends Component {
   }
 
   render() {
-    const {
-      data: { profile },
-      intl: { formatMessage },
-    } = this.props
+    const { data: { profile }, intl } = this.props
 
     const { isMouseOnButton, isMouseOnContent } = this.state
```
Comment (lucasecdb) [inline react/Login.js]: There's no need for these flags on the state. You probably based this on what was being done on the `MiniCart` app, right? I altered the code there to only use the pseudo css selector `:hover`, as it make easier to debug the popup

File: react/components/RecoveryPassword.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+          {isUserBlocked &&
+            <div className="f7 tc bg-washed-red pa2 ma1">
+              {translate('login.userBlocked', intl)}
+            </div>
+          }
+          <div className="pv3">
+            <Input
+              type="password"
+              onChange={this.handleConfirmPassword}
+              placeholder={translate('login.password', intl)}
```
Comment (lucasecdb) [inline react/components/RecoveryPassword.js]: Shouldn't this be something like `login.confirmPassword`? The form looks weird with two fields with a "Password" placeholder

File: react/components/RecoveryPassword.js [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+      })
+        .then(({ data }) => {
+          if (data && data.recoveryPassword) {
+            this.setState({ isLoading: false })
+            this.handleSuccess(data.recoveryPassword)
+            this.handleUserIsBlocked(data.recoveryPassword)
+          }
+        }, err => { console.error(err) })
+    }
+    event.preventDefault()
```
Comment (lucasecdb) [inline react/components/RecoveryPassword.js]: Move this to the top of the function

File: react/components/EmailVerification.js [hunk 5]
```diff
@@ -38,7 +39,9 @@ class EmailVerification extends Component {
         .then(({ data }) => {
           if (data && data.sendEmailVerification) {
             this.setState({ isLoading: false })
-            onStateChange({ step: next })
+            isCreatePassword
+              ? onStateChange({ step: steps.CREATE_PASSWORD, isCreatePassword: false })
```
Comment (andreldsa) [inline react/components/EmailVerification.js]: A little bit strange, you're passing the step being create a password but telling that is NOT (false) creating the password.

File: react/components/RecoveryPassword.js [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+  handleOnSubmit = event => {
+    const { email, recoveryPassword } = this.props
+    const { newPassword, code, confirmPassword } = this.state
+    if (!isValidAccesCode(code)) {
+      this.setState({ isInvalidCode: true })
+    } else if (!isValidPassword(newPassword)) {
+      this.setState({ isInvalidPassword: true })
+    } else if (newPassword !== confirmPassword) {
+      console.log(newPassword, confirmPassword)
```
Comment (andreldsa) [inline react/components/RecoveryPassword.js]: console.log

File: react/components/RecoveryPassword.js [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+  }
+
+  handleUserIsBlocked = status => {
+    status === 'BlockedUser' && this.setState({ isUserBlocked: true })
+  }
+
+  handleOnSubmit = event => {
+    const { email, recoveryPassword } = this.props
+    const { newPassword, code, confirmPassword } = this.state
+    if (!isValidAccesCode(code)) {
```
Comment (andreldsa) [inline react/components/RecoveryPassword.js]: Access*

**Pull request merged**